e2e/utils/trade.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable prettier/prettier */
+import { Page, expect } from '@playwright/test';
+import { navigateTo, waitFor } from './operators';
+import { closeModal, waitModalClose, waitModalOpen } from './modal';
+
+interface TradeConfig {
+  mode: 'buy' | 'sell';
+  source: string;
+  target: string;
+  sourceValue: string;
+  targetValue: string;
+}
+
+const getBalance = (page: Page, token: string) => {
+  return page.getByTestId(`balance-${token}`);
+};
+
+export const testTrade = async (page: Page, config: TradeConfig) => {
```

Comment:
Here the assertion is mixed with the automation. The class should offer methods to selectPair, enter pay, etc. and assertion should be made by the caller by asking the class to fetch the values